### ISSUE TITLE
double clic_unload

### ISSUE THREAD

**Issue by brodieG** [r-lib/cli#319]
I've run into an interesting bug that manifests as:

```
Error in .Call(clic_unload) : NULL value passed as symbol address
```

In essence, what appears to be happening is that if one unloads the `cli` namespace, then `clic_unload` is invoked via `.onUnload`, but then at a later point if the namespace is gc'ed, the entry point is called again while being possibly in a corrupted state.

As I noted I cannot reproduce the actual error simply, but I did modify `cli` to show when it's running `clic_unload`:

```
--- a/R/onload.R
+++ b/R/onload.R
@@ -57,7 +57,11 @@ task_callback <- NULL
   )
 
   # For valgrind: https://github.com/r-lib/cli/issues/311
-  reg.finalizer(asNamespace("cli"), function(x) .Call(clic_unload), TRUE)
+  reg.finalizer(
+      asNamespace("cli"), 
+      function(x) {message("finalizer");.Call(clic_unload)},
+      TRUE
+  )
 
   if (getRversion() >= "3.5.0") {
     `__cli_update_due` <<- .Call(clic_make_timer);
@@ -141,6 +145,7 @@ task_callback <- NULL
   tryCatch(removeTaskCallback(task_callback), error = function(e) NULL)
   tryCatch(cli_progress_cleanup(), error = function(e) NULL)
   tryCatch(ansi_show_cursor(), error = function(e) NULL)
+  message("unload")
   .Call(clic_unload)
 }
```

With that patch I can then create this session transcript:

```
vagrant@vagrant:/vagrant/trunk$ ./bin/R --no-restore --quiet
> loadNamespace('cli')
<environment: namespace:cli>
> unloadNamespace('cli')
unload
> q()
Save workspace image? [y/n/c]: n
finalizer
```

I have not been able to trigger the finalizer simply (e.g. by running `gc(full=TRUE)`) and obviously the above does not show the error, but it does show `clic_unload` being run _after_ the namespace is unloaded, which seems dangerous.  In some of my testing I got a segfault, but again in nothing simple enough to be reproducible, so I am not certain that `cli` is responsible.

The actual error occurred in the context of tests being run with `test_dir`, with code that attempts to unload most loaded packages.

There is no need to address this issue for me as I'm working around it.  Possibly related to #311 and #318.

**Comment by gaborcsardi** [comment]
I think this was fixed in 8587a720c4aacbc4c10d1230029de2149fc7c976

**Comment by brodieG** [comment]
I can confirm my original problem is resolved with the change.  Thanks!
